docs/content/preview/explore/ysql-language-features/pg-extensions/extension-pgvector.md

```diff
@@ -0,0 +1,157 @@
+---
+title: pgvector extension
+headerTitle: pgvector extension
+linkTitle: pgvector
+description: Using the pgvector extension in YugabyteDB
+menu:
+  preview:
+    identifier: extension-pgvector
+    parent: pg-extensions
+    weight: 20
+type: docs
+---
+
+The [pgvector](https://github.com/pgvector/pgvector) PostgreSQL extension allows you to store and query vectors, for use in performing similarity searches.
+
+Note: YugabyteDB support for pgvector does not currently include indexing.
+
+To enable the extension:
+
+```sql
+CREATE EXTENSION vector;
+```
+
+## Create vectors
+
+Create a vector column with 3 dimensions:
+
+```sql
+CREATE TABLE items (id bigserial PRIMARY KEY, embedding vector(3));
+```
+
+Insert vectors:
+
+```sql
+INSERT INTO items (embedding) VALUES ('[1,2,3]'), ('[4,5,6]');
+```
+
+Get the nearest neighbors by L2 distance:
+
+```sql
+SELECT * FROM items ORDER BY embedding <-> '[3,1,2]' LIMIT 5;
+```
+
+The extension also supports inner product (`<#>`) and cosine distance (`<=>`).
+
+Note: `<#>` returns the negative inner product because PostgreSQL only supports `ASC` order index scans on operators.
+
+## Store vectors
+
+Create a new table with a vector column:
+
+```sql
+CREATE TABLE items (id bigserial PRIMARY KEY, embedding vector(3));
+```
+
+Or add a vector column to an existing table:
+
+```sql
+ALTER TABLE items ADD COLUMN embedding vector(3);
+```
+
+Insert vectors:
+
+```sql
+INSERT INTO items (embedding) VALUES ('[1,2,3]'), ('[4,5,6]');
+```
+
+Upsert vectors:
+
+```sql
+INSERT INTO items (id, embedding) VALUES (1, '[1,2,3]'), (2, '[4,5,6]')
+    ON CONFLICT (id) DO UPDATE SET embedding = EXCLUDED.embedding;
+```
+
+Update vectors:
+
+```sql
+UPDATE items SET embedding = '[1,2,3]' WHERE id = 1;
+```
+
+Delete vectors:
+
+```sql
+DELETE FROM items WHERE id = 1;
+```
+
+## Query vectors
+
+Get the nearest neighbors to a vector:
+
+```sql
+SELECT * FROM items ORDER BY embedding <-> '[3,1,2]' LIMIT 5;
+```
+
+Get the nearest neighbors to a row:
+
+```sql
+SELECT * FROM items WHERE id != 1 ORDER BY embedding <-> (SELECT embedding FROM items WHERE id = 1) LIMIT 5;
+```
+
+Get rows within a certain distance:
+
+```sql
+SELECT * FROM items WHERE embedding <-> '[3,1,2]' < 5;
+```
+
+Note: Combine with `ORDER BY` and `LIMIT` to use an index.
```

Comment:
I don't understand. Indexing is not supported right. Is this an alternative?

Reply:
@karthik-ramanathan-3006 can you comment

Reply:
I believe 'ORDER BY' and 'LIMIT' are ways to force the planner to execute the query using indexes.
Since we do not support indexes, this is not applicable to us. 
**Suggestion: We should remove this note.**
Apologies, I missed this note while reviewing.